storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw.go

```diff
@@ -260,3 +272,44 @@ func (c *PrometheusConverter) addSample(sample *prompb.Sample, lbls []prompb.Lab
 	ts.Samples = append(ts.Samples, *sample)
 	return ts
 }
+
+func NewResourceAttributesSetting(otlpCfg config.OTLPConfig) ResourceAttributesSetting {
+	createAttr := func(attributes []string) map[string]struct{} {
+		attr := make(map[string]struct{}, len(attributes))
+		for _, s := range attributes {
+			attr[s] = struct{}{}
+		}
+		return attr
+	}
+
+	if otlpCfg.PromoteAllResourceAttributes {
+		return ResourceAttributesSetting{
+			Action: PromoteAllExceptIgnoreResourceAttributeAction,
+			Attr:   createAttr(otlpCfg.IgnoreResourceAttributes),
+		}
+	}
+
+	return ResourceAttributesSetting{
+		Action: PromoteSpecificResourceAttributeAction,
+		Attr:   createAttr(otlpCfg.PromoteResourceAttributes),
+	}
+}
```

Comment:
This can be made more readable, and by returning a pointer, we can use nil to signal that there's nothing to promote as before this PR:
```suggestion
func NewPromoteResourceAttributes(otlpCfg config.OTLPConfig) *PromoteResourceAttributes {
	if otlpCfg.PromoteAllResourceAttributes {
		attrs := make(map[string]struct{}, len(otlpCfg.IgnoreResourceAttributes))
		for _, s := range otlpCfg.IgnoreResourceAttributes {
			attrs[s] = struct{}{}
		}
		return &PromoteResourceAttributes{
			promoteAll: true,
			attrs:      attrs,
		}
	}

	attrs := make(map[string]struct{}, len(otlpCfg.PromoteResourceAttributes))
	for _, s := range otlpCfg.PromoteResourceAttributes {
		attrs[s] = struct{}{}
	}
	return &PromoteResourceAttributes{
		promoteAll: false,
		attrs:      attrs,
	}
}
```